users/vieting/jobs/fairseq.py

```diff
@@ -204,4 +204,127 @@ def create_dict_ltr(self):
+        :param nbest: number of nbest hypotheses to output, default 1
+        :param w2l_decoder: decoder to use, default "viterbi". Can be "viterbi", "kenlm" or "fairseqlm".
+        :param lm_path: path to language model, default None. Only required if w2l_decoder is "kenlm" or "fairseqlm".
+        :param lm_lexicon: path to lexicon for language model, default None.
+            Only required if w2l_decoder is "kenlm" or "fairseqlm".
+        :param lm_weight: weight of language model, default 2.0
+        :param word_score: word score, default 1.0
+        :param sil_weight: silence weight, default 0.0
+        :param criterion: criterion to use, default "ctc". At the moment, only "ctc" is tested.
+        :param labels: labels to use, default "ltr". At the moment, only "ltr" is tested. 
+            The data_path folder should contain a file with the corresponding file extension.
+        :param post_process: post processing to use, default "letter".
+            Can be "letter", "sentencepiece", "wordpiece", "letter", "silence", "subword_nmt", "_EOW" or "none".
+            (just use "letter" for now)
+        :param max_tokens: maximum number of tokens to decode, default 4000000
+        """
+
+        # inputs
+        self.fairseq_root = fairseq_root
+        self.model_path = model_path
+        self.data_path = data_path
+        self.gen_subset = gen_subset
+        self.nbest = nbest
+
+        assert w2l_decoder in ["viterbi", "kenlm", "fairseqlm"], f"Unknown decoder {w2l_decoder}."
+        self.w2l_decoder = w2l_decoder
+
+        if self.w2l_decoder in ["kenlm", "fairseqlm"]:
+            assert lm_path is not None, "lm_path must be set if w2l_decoder is kenlm or fairseqlm."
+            assert lm_lexicon is not None, "lm_lexicon must be set if w2l_decoder is kenlm or fairseqlm."
+        self.lm_path = lm_path
+        self.lm_lexicon = lm_lexicon
+
+        self.lm_weight = lm_weight
+        self.word_score = word_score
+        self.sil_weight = sil_weight
+
+        if criterion not in ["ctc"]:
+            logging.warning(f"Unknown criterion {criterion}. This might not work.")
+        self.criterion = criterion
+
+        if labels not in ["ltr"]:
+            logging.warning(f"Unknown labels {labels}. This might not work.")
+        self.labels = labels
+
+        if post_process not in ["letter"]:
+            logging.warning(f"Unknown post_process {post_process}. This might not work.")
+        self.post_process = post_process
+
+        self.max_tokens = max_tokens
+
+        # outputs
+        self.out_results = self.output_path("results", directory=True)
+
+        # rqmt
+        self.rqmt = {"time": 6, "mem": 8, "cpu": 1, "gpu": 1, "gpu_mem": 11}
+
+    def tasks(self):
+        yield Task("run", rqmt=self.rqmt)
+
+    def run(self):
+        my_env = os.environ
+        sp.check_call(self._get_run_cmd(), env=my_env)
```

Comment:
I'd maybe print the run command to the log, that might help with debugging sometimes.